src/init/init.ts

```diff
@@ -112,11 +118,7 @@ function getGeneratorDescription(generator: YeomanEnvironment.GeneratorMeta, gen
  * Extract the meaningful name from the full yeoman generator name
  */
 function getDisplayName(generatorName: string) {
-  let nameEnd = generatorName.indexOf(':');
-  if (nameEnd === -1) {
-    nameEnd = generatorName.length;
-  }
-  return generatorName.substring('polymer-init-'.length, nameEnd);
+  return generatorName.replace(/(generator-)?polymer-init-([^:]+)(:.*)?/g, '$2');
```

Comment:
@FredKSchott Good point 👍  I've added comments in 1d21cf5. Let me know if wordsmithing is needed.